cmd/argo/commands/stop.go

```diff
@@ -31,28 +46,75 @@ func NewStopCommand() *cobra.Command {
   argo stop @latest
 `,
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 && !stopArgs.hasSelector() {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
 			ctx, apiClient := client.NewAPIClient()
 			serviceClient := apiClient.NewWorkflowServiceClient()
-			namespace := client.Namespace()
+			stopArgs.namespace = client.Namespace()
 
-			selector, err := fields.ParseSelector(stopArgs.nodeFieldSelector)
-			if err != nil {
-				log.Fatalf("Unable to parse node field selector '%s': %s", stopArgs.nodeFieldSelector, err)
-			}
-
-			for _, name := range args {
-				wf, err := serviceClient.StopWorkflow(ctx, &workflowpkg.WorkflowStopRequest{
-					Name:              name,
-					Namespace:         namespace,
-					NodeFieldSelector: selector.String(),
-					Message:           stopArgs.message,
-				})
-				errors.CheckError(err)
-				fmt.Printf("workflow %s stopped\n", wf.Name)
-			}
+			err := stopWorkflows(ctx, serviceClient, stopArgs, args)
+			errors.CheckError(err)
 		},
 	}
 	command.Flags().StringVar(&stopArgs.message, "message", "", "Message to add to previously running nodes")
 	command.Flags().StringVar(&stopArgs.nodeFieldSelector, "node-field-selector", "", "selector of node to stop, eg: --node-field-selector inputs.paramaters.myparam.value=abc")
+	command.Flags().StringVarP(&stopArgs.labelSelector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
```

Comment:
can I please confirm these descriptions are consistent with other commands?

Reply:
Thanks! Looks like the label selector descriptions from the existing commands are not quite consistent. Here is a summary:

- `argo stop`: Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)
- `argo list`: Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)
- `argo delete`: Selector (label query) to filter on, not including uninitialized ones
- `argo terminate`: Selector (label query) to filter on, not including uninitialized ones
- `argo archive list`: Selector (label query) to filter on, not including uninitialized ones

In the latest commit, I've changed all of them to `Selector (label query) to filter on, not including uninitialized ones, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)`

The field selector does have consistent descriptions.

Let me know if you have any feedback on this. Thanks!